### PR TITLE
Get rid of default predeploy_commands.rb

### PR DESCRIPTION
**Pull request by apurvis**
- rename `target_from_name!` to `get_target_by_name!`
- trivial style changes to cut down on some lines
- fix bug with missing SSH user

there are no functionality changes in here except getting rid of `predeploy_commands.rb`; everything else is just style.

### CODE REVIEW

File: lib/broadside.rb [hunk 1]
```diff
@@ -25,21 +24,15 @@ def self.configure
...
+    raise ArgumentError, "#{config_file} does not exist" unless File.exist?(config_file)
 
+    config.config_file = config_file
     begin
-      config.config_file = config_file
-      load config_file
-    rescue LoadError => e
-      error "Encountered an error loading required configuration file '#{config_file}' !"
-      raise e
+      [config_file, USER_CONFIG_FILE].each { |file| load file if File.exist?(file) }
```
Comment (matl33t) [inline lib/broadside.rb]: config_file should be loaded last, as it has precedence. also, could we add debug statements before each load? Like
```
debug "Loading broadside configuration at #{file}"
```

File: lib/broadside.rb [hunk 2]
```diff
@@ -25,21 +24,15 @@ def self.configure
...
+    rescue LoadError
+      error 'Encountered an error loading broadside configuration'
+      raise
```
Comment (matl33t) [inline lib/broadside.rb]: oh, does this just reraise the same error?

File: lib/broadside.rb (hunk 1 above)
Comment (apurvis) [inline lib/broadside.rb]: oh dip great catch

File: lib/broadside.rb (hunk 2 above)
Comment (apurvis) [inline lib/broadside.rb]: yea

File: lib/broadside.rb (hunk 1 above)
Comment (apurvis) [inline lib/broadside.rb]: (i now officially feel guilty for deleting the spec that checks load order - when i was unifying all the spec contexts that particular spec caused some difficulties and i thought [these lines would force a call order](https://github.com/lumoslabs/broadside/blob/master/spec/broadside_spec.rb#L17-L18) but i just checked and they do not.  😞 )

File: lib/broadside/configuration.rb [hunk 3]
```diff
@@ -25,15 +25,17 @@ class Configuration
     validates_each(:aws) do |record, _, val|
       [:region, :credentials].each { |v| record.errors.add("aws.#{v}") unless val.public_send(v) }
     end
+    validates_each :ssh, allow_nil: true do |record, attr, val|
+      record.errors.add(attr, 'is not a hash') unless val.is_a?(Hash)
+      record.errors.add(attr, 'must contain a user') unless val[:user]
```
Comment (matl33t) [inline lib/broadside/configuration.rb]: no need for the user requirement (someone may have configured a default user in `~/.ssh/config`)

File: lib/broadside/ecs/ecs_manager.rb [hunk 4]
```diff
@@ -63,17 +63,16 @@ def get_running_instance_ips(cluster, family, task_arns = nil)
       end
 
       def get_task_arns(cluster, family, filter = {})
-        opts = {
+        options = {
```
Comment (matl33t) [inline lib/broadside/ecs/ecs_manager.rb]: 🆒

File: spec/broadside/target_spec.rb [hunk 5]
```diff
@@ -30,24 +28,21 @@
     it_behaves_like 'valid_configuration?', true,  env_files: ['file', 'file2']
 
     it_behaves_like 'valid_configuration?', true,  command: nil
-    it_behaves_like 'valid_configuration?', true,  command: ['bundle', 'exec', 'resque:work']
-    it_behaves_like 'valid_configuration?', false, command: 'bundle exec rails s'
+    it_behaves_like 'valid_configuration?', true,  command: %w(do something)
```
Comment (apurvis) [inline spec/broadside/target_spec.rb]: FWIW i am super annoyed that you can't use `let(:command) { %w(do something) }` in an `it_behaves_like` expression.  oh well.

File: lib/broadside/configuration.rb (hunk 3 above)
Comment (apurvis) [inline lib/broadside/configuration.rb]: would the command then be `ssh @123.123.123.123`?  wouldn't you need to ditch the `@`?

Comment (matl33t) [inline lib/broadside/configuration.rb]: good point. i'll address the `@` with https://github.com/lumoslabs/broadside/issues/11

**Pull request merged**